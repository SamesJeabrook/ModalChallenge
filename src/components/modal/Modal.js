import React from "react";
import "./modal.less";
import { getModalSize } from "./getModalSize";

export default class Modal extends React.PureComponent {

  constructor(props){
    super(props);
    this.closeModal = this.closeModal.bind(this);
    this.onEscape = this.onEscape.bind(this);
  }

  closeModal(){
      const {
        onClose,
        modalName,
        canPassDataOnClose,
      } = this.props;
      const onCloseValues = canPassDataOnClose ? {
          message: 'You have closed welcome modal. Message from call back. I will hide in few seconds.'
      } : {};
    onClose && onClose(onCloseValues, modalName);
  };

  onEscape(event){
    if(event.keyCode === 27) {
      this.closeModal();
    }
  }

  componentDidMount(){
    document.addEventListener("keydown", this.onEscape, false);
  }
  componentWillUnmount(){
    document.removeEventListener("keydown", this.onEscape, false);
  }

  render() {
    const {
      children,
      isModalOpen,
      closeButtonLabel,
      modalSize,
      modalTitle,
      modalName,
      autoClose
    } = this.props;

    if(autoClose){
      setTimeout(this.closeModal, 5000);
    }

    const activeModalClassName = isModalOpen ? "show-modal" : "";
    const modalSizeClassName = getModalSize(modalSize);
    return (
      <div className={`modal ${activeModalClassName} ${modalName}`}>
        <section
          className={`modal-main ${activeModalClassName} ${modalSizeClassName}`}
        >
          <div className="modal-header-wrap">
            <h1 className="modal-title">{modalTitle}</h1>
          </div>
          <div className="modal-content">
            {children}
          </div>
          { !autoClose && <footer className="modal-footer">
            <button className="button button_primary" onClick={this.closeModal}>{closeButtonLabel}</button>
          </footer>
          }
        </section>
      </div>
    );
  }
}
