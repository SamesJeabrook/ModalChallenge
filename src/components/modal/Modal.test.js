import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Modal from './Modal';

test('validate Modal componet', function(){
    const html = (<Modal/>);
    const componentTree = renderer.create(html).toJSON();

    expect(componentTree).toMatchSnapshot();
})

test('It add show-modal class when modal opens', function(){
    const modalComponent = shallow(<Modal isModalOpen={true} />);
    expect(modalComponent.find('div.modal').hasClass('show-modal')).toBeTruthy();
})

test('It remove show-modal class when modal opens', function(){
    const modalComponent = shallow(<Modal isModalOpen={false} />);
    expect(modalComponent.find('div.modal').hasClass('show-modal')).toBeFalsy();
})

test('It add modal size', function(){
    const modalComponent = shallow(<Modal  modalSize='small' />);
    expect(modalComponent.find('.modal-main').hasClass('small')).toBeTruthy();
})

test('It add modal size for large', function(){
    const modalComponent = shallow(<Modal  modalSize='large' />);
    expect(modalComponent.find('.modal-main').hasClass('large')).toBeTruthy();
})

test('It add modal size default if it is not small or large', function(){
    const modalComponent = shallow(<Modal modalSize='some' />);
    expect(modalComponent.find('.modal-main').hasClass('default')).toBeTruthy();
})