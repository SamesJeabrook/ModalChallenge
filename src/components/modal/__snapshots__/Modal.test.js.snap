// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Modal componet 1`] = `
<div
  className="modal  undefined"
>
  <section
    className="modal-main  default"
  >
    <div
      className="modal-header-wrap"
    >
      <h1
        className="modal-title"
      />
    </div>
    <div
      className="modal-content"
    />
    <footer
      className="modal-footer"
    >
      <button
        className="button button_primary"
        onClick={[Function]}
      />
    </footer>
  </section>
</div>
`;
